core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -3070,10 +3071,10 @@ public void run(Reactor session) throws Exception {
                     items.put(item.getName(), item);
                     loadedNames.add(item.getName());
                 }
-            });
+            }));
         }
 
-        g.requires(JOB_LOADED).attains(COMPLETED).add("Cleaning up obsolete items deleted from the disk", new Executable() {
+        g.requires(loadJobs.toArray(new Handle[loadJobs.size()])).attains(COMPLETED).add("Cleaning up obsolete items deleted from the disk", new Executable() {
```

Comment:
Seems hackish—really what we needed here is another milestone after `JOB_LOADED`, right? But I do not understand this issue very well.

Reply:
I don't see why someone would want to depend on a milestone taking place in between job loading and obsolete job clean up - that is the current `JOB_LOADED` milestone. That's why I'm moving it after the obsolete job clean up has been done.  Code-wise, I agree it would be simpler to read with a new milestone, however I don't think there is any benefit to make such a milestone public.